pkg/operator/configobserver/apiserver/observe_tlssecurityprofile.go

```diff
@@ -24,23 +24,47 @@ type APIServerLister interface {
 	PreRunHasSynced() []cache.InformerSynced
 }
 
-// ObserveTLSSecurityProfile observes APIServer.Spec.TLSSecurityProfile field and sets
-// the ServingInfo.MinTLSVersion, ServingInfo.CipherSuites fields
-func ObserveTLSSecurityProfile(genericListers configobserver.Listers, recorder events.Recorder, existingConfig map[string]interface{}) (map[string]interface{}, []error) {
-	var (
-		minTlSVersionPath = []string{"servingInfo", "minTLSVersion"}
-		cipherSuitesPath  = []string{"servingInfo", "cipherSuites"}
-	)
+type tlsSecurityObserver struct {
+	minTlSVersionPath []string
+	cipherSuitesPath  []string
+}
+
+var (
+	// minTLSVersionPath provides the path to access minimum serving TLS version in
+	// the observed config based on whether apiServerArguments field should be used
+	minTLSVersionPath = map[bool][]string{
+		false: {"servingInfo", "minTLSVersion"},
+		true:  {"apiServerArguments", "tls-min-version"},
+	}
+	// cipherSuitesPath provides the path to serving cipher suites in the observed
+	// config based on whether apiServerArguments field should be used
+	cipherSuitesPath = map[bool][]string{
+		false: {"servingInfo", "cipherSuites"},
+		true:  {"apiServerArguments", "tls-cipher-suites"},
+	}
+)
+
+// NewTLSSSecurityProfileObserver creates and observer for the APIServer.Spec.TLSSecurityProfile field
+// which sets the ServingInfo.MinTLSVersion and ServingInfo.CipherSuites fields values in the observed
+// configuration (or their apiServerArguments mirrors)
+func NewTLSSSecurityProfileObserver(useAPIServerArguments bool) configobserver.ObserveConfigFunc {
```

Comment:
Just wondering if we could do something like the following for minimal disruption to components that use library-go  . So we could preserve the old interface, extract the common logic and provide a new function.


```
func ObserveTLSSecurityProfile(genericListers configobserver.Listers, recorder events.Recorder, existingConfig map[string]interface{}) (map[string]interface{}, []error) {
	var (
		minTlSVersionPath = []string{"servingInfo", "minTLSVersion"}
		cipherSuitesPath  = []string{"servingInfo", "cipherSuites"}
	)
        // here we could call the common function passing the old `minTlSVersionPath` and `cipherSuitesPath`

}
```

```
func NewObserveTLSSecurityProfile(genericListers configobserver.Listers, recorder events.Recorder, existingConfig map[string]interface{}) (map[string]interface{}, []error) {
	var (
		minTlSVersionPath = []string{"apiServerArguments", "tls-min-version"}
		cipherSuitesPath  = []string{"apiServerArguments", "tls-cipher-suites"}
	)
        // here we would call the common function passing different `minTlSVersionPath` and `cipherSuitesPath` 
       //  TODO: do we have to clean up the existingConfig - remove the old entries (`servingInfo`) ?

}
```

In the future releases we could just simply remove  `ObserveTLSSecurityProfile` and rename `NewObserveTLSSecurityProfile`

Reply:
Definitely makes more sense from the perspective of the consumer of these functions, I'll make it so.